app/src/main/java/com/github/se/eventradar/viewmodel/qrCode/ScanFriendQrViewModel.kt

```diff
@@ -121,16 +211,18 @@ constructor(
     return updateResult is Resource.Success
   }
 
-  fun resetNavigationEvent() {
-    _uiState.value = _uiState.value.copy(action = Action.None)
+  private fun changeAction(action: Action) {
+    _uiState.value = uiState.value.copy(action = action)
+    if (action == Action.NavigateToNextScreen) {
+      navigationActions.navController.navigate(
```

Comment:
any calls to the navigation actions should be in the UI state. This is how all the other screens operate and it means you don't have to worry about assisted hilt injection with the navigation actions

Reply:
I made these changes because the coach told me to do so, read discoord to understand why it was causing race conditions